DBAL/TracingStatement.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingDoctrineDBALBundle\DBAL;
+
+use Doctrine\DBAL\Driver\Statement;
+use IteratorAggregate;
+
+/**
+ * @SuppressWarnings(PHPMD.TooManyPublicMethods)
+ * @implements IteratorAggregate<Statement>
+ */
+final class TracingStatement implements IteratorAggregate, Statement
+{
+    /**
+     * @var Statement<Statement>
+     */
+    private $statement;
+    private $sql;
+    private $spanFactory;
+    private $username;
+    /**
+     * @var array<mixed>
+     */
+    private $params = [];
+
+    /**
+     * @param Statement<Statement> $statement
```

Comment:
I don't understand this comment. The type hint is clear to me.

Reply:
this is satisfying PHPStan